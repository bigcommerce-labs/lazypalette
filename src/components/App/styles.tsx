import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
