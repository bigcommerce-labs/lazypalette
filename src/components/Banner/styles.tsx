import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
