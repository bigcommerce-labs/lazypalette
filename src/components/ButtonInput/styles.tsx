import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
