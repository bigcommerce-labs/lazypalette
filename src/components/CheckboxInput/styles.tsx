import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
