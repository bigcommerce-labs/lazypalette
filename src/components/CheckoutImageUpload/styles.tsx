import styled from 'styled-components';

export const UploadButtonLabel = styled.label`
    height: 2rem;
    border-radius: 2px;
    border: 1px solid ${({ theme }) => theme.colors.primary};
    padding: 0 0.75em;
    font-size: 0.75rem;
    cursor: pointer;
    display: inline-block;
    color: ${({ theme }) => theme.colors.primary};
    margin-right: 1rem;
`;

export const CheckoutImageUploadWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImageInputWrapper = styled.div`
    margin: 1rem 0;
`;

export const CheckoutImagePreviewWrapper = styled.div``;

export const CheckoutImageUploadError = styled.p`
    color: ${({ theme }) => theme.colors.error};
`;

export const CheckoutImageRemoveLink = styled.a`
    color: ${({ theme }) => theme.colors.error};
`;
