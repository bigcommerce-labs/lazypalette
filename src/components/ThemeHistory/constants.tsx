export const enum Messages {
    Reset = 'This theme has unpublished changes. Do you want to proceed?',
}

export const PostLaunchTypeLabels = {
    default: 'Default Settings',
    installed: 'Published',
    saved: 'Saved',
    upgrade: 'Update',
};

export const PreLaunchTypeLabels = {
    default: 'Default Settings',
    installed: 'Saved',
    saved: 'Saved',
    upgrade: 'Update',
};
