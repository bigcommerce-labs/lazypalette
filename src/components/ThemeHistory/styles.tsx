import styled from 'styled-components';

export const List = styled.ul`
    cursor: pointer;
    list-style: none;
    padding: 0;
    margin: 0 1rem 0 1.5rem;
    width: 20rem;
`;

export const HistoryEntry = styled.li`
    margin: 0 0 2rem 0;
`;

export const EntryActive = styled.div`
    background: ${({theme}) => theme.colors.success};
    border-radius: 16px;
    bottom: 0;
    display: inline-block;
    height: 1.25rem;
    left: 0.5rem;
    margin-top: -1rem;
    position: relative;
    top: 0.25rem;
    width: 1.25rem;

    :after {
        content: '';
        position: absolute;
        width: .5rem;
        height: .25rem;
        background: transparent;
        top: .3125rem;
        left: .3125rem;
        border: 2px solid ${({theme}) => theme.colors.empty};
        border-top: none;
        border-right: none;
        transform: rotate(-45deg);
    }
`;

EntryActive.defaultProps = {
    theme: {
        colors: {
            empty: '#FFFFFF',
            success: '#AAAAAA',
        },
    },
};

export const EntryDescription = styled.div`
    color: ${({theme}) => theme.colors.primaryText };
    line-height: 1.5rem;
    margin: 0 0 .25rem 0;
    position: relative;
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold };
`;

EntryDescription.defaultProps = {
    theme: {
        colors: {
            primaryText: '#FFFFFF',
        },
        typography: {
            fontWeight: {
                bold: 600,
            },
        },
    },
};

export const EntryVersion = styled.div`
    color: ${({theme}) => theme.colors.secondaryText };
    line-height: 1.5rem;
`;

EntryVersion.defaultProps = {
    theme: {
        colors: {
            secondaryText: '#FFFFFF',
        },
    },
};
