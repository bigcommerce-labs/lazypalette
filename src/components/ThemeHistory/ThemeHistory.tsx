import React, { PureComponent, SFC } from 'react';
import { connect } from 'react-redux';
import { withRouter, Route, RouteComponentProps } from 'react-router-dom';
import { Dispatch } from 'redux';

import { loadTheme } from '../../actions/theme';
import { State } from '../../reducers/reducers';
import { ThemeVariationHistory, ThemeVariationHistoryEntry } from '../../reducers/theme';
import Draggable from '../Draggable/Draggable';
import ExpandableMenu from '../ExpandableMenu/ExpandableMenu';
import ConfirmModal from '../Modal/ConfirmModal/ConfirmModal';
import { appRoutes } from '../Routes/Routes';

import { Messages, PostLaunchTypeLabels, PreLaunchTypeLabels } from './constants';
import { EntryActive, EntryDescription, EntryVersion, HistoryEntry, List } from './styles';

interface ThemeHistoryProps extends RouteComponentProps<{}> {
    configurationId: string;
    isChanged: boolean;
    isPrelaunchStore: boolean;
    position: { x: number, y: number };
    timeZone?: string; // for testing to ensure consistent snapshots
    variationHistory: ThemeVariationHistory;
    loadTheme(configurationId: string, variationId: string): Dispatch<State>;
}

interface ThemeHistoryState {
    confirmData: {
        configurationId: string;
        variationId: string;
    };
    isConfirmOpen: boolean;
}

export class ThemeHistory extends PureComponent<ThemeHistoryProps, ThemeHistoryState> {
    readonly state: ThemeHistoryState = {
        confirmData: {
            configurationId: '',
            variationId: '',
        },
        isConfirmOpen: false,
    };

    openModal = (variationId: string, configurationId: string) => this.setState({
        confirmData: {
            configurationId,
            variationId,
        },
        isConfirmOpen: true,
    });

    handleModalCancel = () => this.setState({
        isConfirmOpen: false,
    });

    handleModalConfirm = () => {
        const { configurationId, variationId } = this.state.confirmData;

        this.setState({
            isConfirmOpen: false,
        }, () => {
            this.props.loadTheme(variationId, configurationId);
        });
    };

    handleEntrySelect = (variationId: string, configurationId: string) => {
        if (this.props.isChanged) {
            this.openModal(variationId, configurationId);
        } else {
            this.props.loadTheme(variationId, configurationId);
        }
    };

    getEntryDescription(entry: ThemeVariationHistoryEntry) {
        const date = new Date(entry.timestamp);
        const localeOptions = {
            day: 'numeric',
            hour: 'numeric',
            minute: 'numeric',
            month: 'long',
            timeZone: this.props.timeZone,
            year: 'numeric',
        };

        const dateString = date.toLocaleDateString('en', localeOptions);

        const typeLabels = this.props.isPrelaunchStore ? PreLaunchTypeLabels : PostLaunchTypeLabels;
        const typeString = typeLabels[entry.type] || entry.type;

        return this.showDate(entry.type) ? `${typeString} ${dateString}` : typeString;
    }

    showDate = (type: string) => {
        return type === 'installed' || type === 'saved';
    };

    render() {
        const { match, position, variationHistory} = this.props;
        const { isConfirmOpen } = this.state;

        return (
            <>
                <Draggable position={position}>
                    <ExpandableMenu title="History" back={match.url}>
                        <List>
                            {variationHistory.map(entry => (
                                <HistoryEntry
                                    onClick={() =>
                                        this.handleEntrySelect(entry.variationId, entry.configurationId)}
                                    key={entry.configurationId}>
                                    <EntryDescription>
                                        {this.getEntryDescription(entry)}
                                        {entry.configurationId === this.props.configurationId &&
                                            <EntryActive/>}
                                    </EntryDescription>
                                    <EntryVersion>
                                        Version {entry.displayVersion}
                                    </EntryVersion>
                                </HistoryEntry>
                            ))}
                        </List>
                    </ExpandableMenu>
                </Draggable>
                {isConfirmOpen &&
                    <ConfirmModal
                        body={Messages.Reset}
                        primaryAction={this.handleModalConfirm}
                        secondaryAction={this.handleModalCancel}
                        title="Theme Change Warning"
                    />
                }
            </>
        );
    }
}

const RoutedThemeHistory: SFC<ThemeHistoryProps> = props => (
    <Route
        path={appRoutes.history.path}
        exact
        render={() => <ThemeHistory {...props}/>}
    />
);

const mapStateToProps = (state: State) => ({
    configurationId: state.theme.initialConfigurationId,
    isChanged: state.theme.isChanged,
    isPrelaunchStore: state.merchant.isPrelaunchStore,
    variationHistory: state.theme.variationHistory,
});

const mapDispatchToProps = {
    loadTheme,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RoutedThemeHistory));
