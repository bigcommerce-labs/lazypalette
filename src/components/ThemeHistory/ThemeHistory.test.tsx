import { shallow } from 'enzyme';
import React from 'react';

import { HistoryEntry } from './styles';
import { ThemeHistory } from './ThemeHistory';

describe('ThemeHistory', () => {
    const history = [{
        configurationId: 'config123',
        displayVersion: 'Version 1.0',
        themeId: 'meow1',
        themeName: 'Meowcat Theme',
        timestamp: '2018-08-17T18:13:42Z',
        type: 'installed',
        variationId: 'variation123',
        variationName: 'Variation Meow',
        versionId: 'version123',
    },
    {
        configurationId: 'config456',
        displayVersion: 'Version 2.0',
        themeId: 'woof1',
        themeName: 'Doge Theme',
        timestamp: '2015-08-13T12:13:42Z',
        type: 'saved',
        variationId: 'variation456',
        variationName: 'Variation Woof',
        versionId: 'version456',
    },
    {
        configurationId: 'congig789',
        displayVersion: 'Version 0.0',
        themeId: 'hiss1',
        themeName: 'Snake Theme',
        timestamp: '2015-08-10T12:13:42Z',
        type: 'default',
        variationId: 'variation789',
        variationName: 'Variation Hiss',
        versionId: 'version789',
    }];

    const mockProp: any = jest.fn();
    const routeProps = {
        history: mockProp,
        location: mockProp,
        match: mockProp,
        staticContext: mockProp,
    };

    let mockCreateNotification: any;
    let mockLoadTheme: any;

    beforeEach(() => {
        mockCreateNotification = jest.fn();
        mockLoadTheme = jest.fn().mockResolvedValue(true);
    });

    describe('when rendering', () => {
        describe('when it has history items', () => {
            describe('when an item is currently active', () => {
                it('renders correctly', () => {
                    const themeHistory = shallow(
                        <ThemeHistory
                            configurationId={history[0].configurationId}
                            isChanged={false}
                            isPrelaunchStore={false}
                            position={{ x: 5, y: 10 }}
                            createNotification={mockCreateNotification}
                            loadTheme={mockLoadTheme}
                            timeZone="UTC"
                            variationHistory={history}
                            {...routeProps}/>);

                    expect(themeHistory).toMatchSnapshot();
                });
            });

            describe('when in prelaunch mode', () => {
                it('renders correctly', () => {
                    const themeHistory = shallow(
                        <ThemeHistory
                            configurationId={history[0].configurationId}
                            isChanged={false}
                            isPrelaunchStore={true}
                            position={{ x: 5, y: 10 }}
                            createNotification={mockCreateNotification}
                            loadTheme={mockLoadTheme}
                            timeZone="UTC"
                            variationHistory={history}
                            {...routeProps}/>);

                    expect(themeHistory).toMatchSnapshot();
                });
            });

            describe('when no items are currently active', () => {
                it('renders correctly', () => {
                    const themeHistory = shallow(
                        <ThemeHistory
                            configurationId={'blaaaah'}
                            isChanged={false}
                            isPrelaunchStore={false}
                            position={{ x: 5, y: 10 }}
                            createNotification={mockCreateNotification}
                            loadTheme={mockLoadTheme}
                            timeZone="UTC"
                            variationHistory={history}
                            {...routeProps}/>);

                    expect(themeHistory).toMatchSnapshot();
                });
            });
        });

        describe('when it has no history items', () => {
            it('renders correctly', () => {
                const themeHistory = shallow(
                    <ThemeHistory
                        configurationId={history[0].configurationId}
                        isChanged={false}
                        isPrelaunchStore={false}
                        position={{ x: 5, y: 10 }}
                        createNotification={mockCreateNotification}
                        loadTheme={mockLoadTheme}
                        timeZone="UTC"
                        variationHistory={[]}
                        {...routeProps}/>);

                expect(themeHistory).toMatchSnapshot();
            });
        });

        describe('when the modal is open', () => {
            it('renders correctly', () => {
                const themeHistory = shallow(
                    <ThemeHistory
                        configurationId={history[0].configurationId}
                        isChanged={true}
                        isPrelaunchStore={false}
                        position={{ x: 5, y: 10 }}
                        createNotification={mockCreateNotification}
                        loadTheme={mockLoadTheme}
                        timeZone="UTC"
                        variationHistory={history}
                        {...routeProps}/>);

                themeHistory.setState({
                    confirmData: {
                        configurationId: history[0].configurationId,
                        variationId: history[0].variationId,
                    },
                    isConfirmOpen: true,
                });
                expect(themeHistory).toMatchSnapshot();
            });
        });
    });

    describe('when an entry is clicked', () => {
        describe('when there are no unsaved changes', () => {
            describe('when loadTheme succeeds', () => {
                it('calls loadTheme', () => {
                    const themeHistory = shallow(
                        <ThemeHistory
                            configurationId={'blaaaah'}
                            isChanged={false}
                            isPrelaunchStore={false}
                            position={{ x: 5, y: 10 }}
                            createNotification={mockCreateNotification}
                            loadTheme={mockLoadTheme}
                            variationHistory={history}
                            {...routeProps}/>);

                    themeHistory.find(HistoryEntry).first().simulate('click');

                    expect(mockLoadTheme).toHaveBeenCalled();
                });
            });

            describe('when loadTheme fails', () => {
                it('creates a notification', done => {
                    mockLoadTheme.mockReset();
                    mockLoadTheme.mockResolvedValue({ error: true });

                    const themeHistory = shallow(
                        <ThemeHistory
                            configurationId={'blaaaah'}
                            isChanged={false}
                            isPrelaunchStore={false}
                            position={{ x: 5, y: 10 }}
                            createNotification={mockCreateNotification}
                            loadTheme={mockLoadTheme}
                            variationHistory={history}
                            {...routeProps}/>);

                    themeHistory.find(HistoryEntry).first().simulate('click');

                    expect(mockLoadTheme).toHaveBeenCalled();
                    setTimeout(() => {
                        expect(mockCreateNotification).toHaveBeenCalled();
                        done();
                    });
                });
            });
        });

        describe('when unsaved changes exist', () => {
            it('does not call loadTheme and opens a modal', () => {
                const themeHistory = shallow(
                    <ThemeHistory
                        configurationId={'blaaaah'}
                        isChanged={true}
                        isPrelaunchStore={false}
                        position={{ x: 5, y: 10 }}
                        createNotification={mockCreateNotification}
                        loadTheme={mockLoadTheme}
                        variationHistory={history}
                        {...routeProps}/>);

                themeHistory.find(HistoryEntry).first().simulate('click');

                expect(mockLoadTheme).not.toHaveBeenCalled();
                expect(themeHistory.state()).toEqual({
                    confirmData: {
                        configurationId: history[0].configurationId,
                        variationId: history[0].variationId,
                    },
                    isConfirmOpen: true,
                });
            });
        });
    });

    describe('when the modal is open', () => {
        let themeHistory: any;

        beforeEach(() => {
            themeHistory = shallow(
                <ThemeHistory
                    configurationId={'blaaaah'}
                    isChanged={true}
                    isPrelaunchStore={false}
                    position={{ x: 5, y: 10 }}
                    createNotification={mockCreateNotification}
                    loadTheme={mockLoadTheme}
                    variationHistory={history}
                    {...routeProps}/>);

            themeHistory.setState({
                confirmData: {
                    configurationId: history[0].configurationId,
                    variationId: history[0].variationId,
                },
                isConfirmOpen: true,
            });
        });

        describe('when the cancel handler is called', () => {
            it('closes the modal and does not call loadTheme', () => {
                themeHistory.instance().handleModalCancel();

                expect(mockLoadTheme).not.toHaveBeenCalled();
            });
        });

        describe('when the confirmation handler is called', () => {
            it('closes the modal and calls loadTheme', () => {
                themeHistory.instance().handleModalConfirm();

                expect(mockLoadTheme).toHaveBeenCalledWith(history[0].variationId, history[0].configurationId);
            });
        });
    });
});
