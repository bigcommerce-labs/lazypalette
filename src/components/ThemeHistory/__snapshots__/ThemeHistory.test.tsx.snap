// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeHistory when rendering when it has history items when an item is currently active renders correctly 1`] = `
<React.Fragment>
  <Draggable
    position={
      Object {
        "x": 5,
        "y": 10,
      }
    }
  >
    <ExpandableMenu
      title="History"
    >
      <styled.ul>
        <styled.li
          key="config123"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Published August 17, 2018, 6:13 PM
            <styled.div
              theme={
                Object {
                  "colors": Object {
                    "empty": "#FFFFFF",
                    "success": "#AAAAAA",
                  },
                }
              }
            />
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 1.0
          </styled.div>
        </styled.li>
        <styled.li
          key="config456"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Saved August 13, 2015, 12:13 PM
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 2.0
          </styled.div>
        </styled.li>
      </styled.ul>
    </ExpandableMenu>
  </Draggable>
</React.Fragment>
`;

exports[`ThemeHistory when rendering when it has history items when no items are currently active renders correctly 1`] = `
<React.Fragment>
  <Draggable
    position={
      Object {
        "x": 5,
        "y": 10,
      }
    }
  >
    <ExpandableMenu
      title="History"
    >
      <styled.ul>
        <styled.li
          key="config123"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Published August 17, 2018, 6:13 PM
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 1.0
          </styled.div>
        </styled.li>
        <styled.li
          key="config456"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Saved August 13, 2015, 12:13 PM
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 2.0
          </styled.div>
        </styled.li>
      </styled.ul>
    </ExpandableMenu>
  </Draggable>
</React.Fragment>
`;

exports[`ThemeHistory when rendering when it has no history items renders correctly 1`] = `
<React.Fragment>
  <Draggable
    position={
      Object {
        "x": 5,
        "y": 10,
      }
    }
  >
    <ExpandableMenu
      title="History"
    >
      <styled.ul />
    </ExpandableMenu>
  </Draggable>
</React.Fragment>
`;

exports[`ThemeHistory when rendering when the modal is open renders correctly 1`] = `
<React.Fragment>
  <Draggable
    position={
      Object {
        "x": 5,
        "y": 10,
      }
    }
  >
    <ExpandableMenu
      title="History"
    >
      <styled.ul>
        <styled.li
          key="config123"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Published August 17, 2018, 6:13 PM
            <styled.div
              theme={
                Object {
                  "colors": Object {
                    "empty": "#FFFFFF",
                    "success": "#AAAAAA",
                  },
                }
              }
            />
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 1.0
          </styled.div>
        </styled.li>
        <styled.li
          key="config456"
          onClick={[Function]}
        >
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "primaryText": "#FFFFFF",
                },
                "typography": Object {
                  "fontWeight": Object {
                    "bold": 600,
                  },
                },
              }
            }
          >
            Saved August 13, 2015, 12:13 PM
          </styled.div>
          <styled.div
            theme={
              Object {
                "colors": Object {
                  "secondaryText": "#FFFFFF",
                },
              }
            }
          >
            Version 
            Version 2.0
          </styled.div>
        </styled.li>
      </styled.ul>
    </ExpandableMenu>
  </Draggable>
  <ConfirmModal
    body="This theme has unpublished changes. Do you want to proceed?"
    primaryAction={[Function]}
    secondaryAction={[Function]}
    title="Theme Change Warning"
  />
</React.Fragment>
`;
